src/browser/components/LoadingAnimation/LoadingAnimation.jsx

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+import useAnimationEnd from '../../hooks/useAnimationEnd.js';
+
+import LoadingIcon from './LoadingIcon.jsx';
+
+const LOADING_STATE = {
+  INITIALIZING: 'initializing', // animation graphics are hidden
+  LOADING: 'loading', // animation graphics fade in and animate
+  LOADED: 'loaded', // animation graphics fade out
+  COMPLETE: 'complete', // animation graphics are removed from the DOM
+};
+
+const ANIMATION_COMPLETION_DELAY = 500;
+
+function LoadingAnimation({
+  loading = false,
+  darkMode = false,
+  onLoadAnimationComplete = () => {}} // eslint-disable-line no-empty-function
```

Comment:
Can we just make this optional or `null` or `undefined` and just do a null check instead of executing an empty function?

Reply:
Yup. I've seen this done all kinds of ways, but not set on one in particular. :)

Reply:
I'm 0/5 on this (either way is fine to me), but if we decide it is ok to have an empty-function we should remove the linting rule.

Reply:
Still looking to have this changed. 3/5, since we're violating the linting rule, and I think a null check is a more common practice.